tests/conftest.py

```diff
@@ -37,13 +39,26 @@ def deactive_venv():
     old_os_path = os.environ.get("PATH", "")
     old_prefix = sys.prefix
     old_path = sys.path
+    old_pythonpath = os.environ.get("PYTHONPATH", None)
+    old_os_venv: Optional[str] = os.environ.get("VIRTUAL_ENV", None)
+    old_working_set = pkg_resources.working_set
 
     yield
 
     os.environ["PATH"] = old_os_path
     sys.prefix = old_prefix
     sys.path = old_path
-    pkg_resources.working_set = pkg_resources.WorkingSet._build_master()
+    pkg_resources.working_set = old_working_set
+    # Restore PYTHONPATH
+    if "PYTHONPATH" in os.environ:
+        if old_pythonpath is not None:
+            os.environ["PYTHONPATH"] = old_pythonpath
+        else:
+            del os.environ["PYTHONPATH"]
+    if old_os_venv is not None:
+        os.environ["VIRTUAL_ENV"] = old_os_venv
```

Comment:
Shouldn't we also unset it if required?

Reply:
I guess so. I copied this from core so I'll apply the change on my PR there as well. Same for the `PYTHON_PATH` change requested above. @wouterdb rerequesting your review for this change.

Reply:
Done

Reply:
This magic is too dark to know what is going to happen by just reading the code, so let's see what gives.